domain/upgrade/state/state.go

```diff
@@ -299,3 +299,38 @@ func (st *State) ActiveUpgrade(ctx context.Context) (string, error) {
 	})
 	return activeUpgrade, errors.Trace(err)
 }
+
+func (st State) SetDBUpgradeCompleted(ctx context.Context, upgradeUUID string) error {
+	db, err := st.DB()
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	getUpgradeDBCompletedQuery := "SELECT db_completed_at AS &Info.* FROM upgrade_info WHERE uuid = $M.info_uuid"
+	getUpgradeDBCompletedStmt, err := sqlair.Prepare(getUpgradeDBCompletedQuery, Info{}, sqlair.M{})
+	if err != nil {
+		return errors.Annotatef(err, "preparing %q", getUpgradeDBCompletedQuery)
+	}
+
+	completeDBUpgradeQuery := "UPDATE upgrade_info SET db_completed_at = DATETIME('now') WHERE uuid = $M.info_uuid"
```

Comment:
I would not do a separate select before attempting to update.
- Add `where db_completed_at IS NULL` to the update statement.
- Check the `RowsAffected` result from running the update.
- If there are no rows affected, return an error saying that there is no upgrade record with a pending DB upgrade.

worker/changestreampruner/worker.go has an example of getting a `sql.Outcome` and checking the result.